ngMax/index.js

```diff
@@ -0,0 +1,45 @@
+define(['angular'], function (angular) {
+  'use strict';
+
+  return angular.module('app.ng-max', []).directive('ngMax', function () {
+    return {
+      restrict: 'A',
+      require: 'ngModel',
+      link: function (scope, element, attr, ctrl) {
+
+        scope.initialize = function () {
+          scope.$watch(attr.ngMax, function () {
+            ctrl.$setViewValue(ctrl.$viewValue);
+          });
+
+          ctrl.$parsers.push( scope.maxValidator );
+          ctrl.$formatters.push( scope.maxValidator );
+        };
+
+        /**
+         * Returns the value and sets validity in true if value didn't surpass the upper limit.
+         * Otherwise sets validity in false and returns undefined.
+         */
+        scope.maxValidator = function (value) {
+          var max = scope.$eval(attr.ngMax) || Infinity;
+          if (!scope.isValidable(value) && value > max) {
+              ctrl.$setValidity('ngMax', false);
```

Comment:
Will it be better to use `max` as validation key? As an example angular `ng-minlength` directive uses `minlength` key.